clang/include/clang/Sema/Sema.h

```diff
@@ -15012,6 +14992,12 @@ class Sema final : public SemaBase {
     return hasAcceptableDefinition(D, &Hidden, Kind);
   }
 
+  /// Try to parse the conditional expression attached to an effect attribute
+  /// (e.g. 'nonblocking'). (c.f. Sema::ActOnNoexceptSpec). Return an empty
+  /// optional on error.
+  std::optional<FunctionEffectMode>
+  ActOnEffectExpression(Expr *CondExpr, StringRef AttributeName);
```

Comment:
Earlier I moved this in Sema to be in the SemaType.cpp section, where the implementation remains, to be next to `handleNonBlockingNonAllocatingTypeAttr()`.

Reply:
Yeah, that seems fine.